### PR TITLE
basic setup of podNetwork disruption tests

### PR DESCRIPTION
**Pull request by deads2k**
adding the skeleton of a new disruption test.

### CODE REVIEW

File: pkg/invariants/network/disruptionpodnetwork/pod-network-target-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        network.openshift.io/disruption-target: pod-network
+        network.openshift.io/disruption-actor: target
+    spec:
+      containers:
+        - command:
+            - /agnhost
+            - netexec
+            - --http-port=8080
+            - --delay-shutdown=30
+          image: registry.k8s.io/e2e-test-images/agnhost:2.43
```
Comment (JacobTanenbaum) [inline pkg/invariants/network/disruptionpodnetwork/pod-network-target-deployment.yaml:29]: Should it be mentioned that the image gets overridden as we do above for replicas?

File: pkg/defaultinvariants/types.go [hunk 2]
```diff
@@ -70,6 +71,7 @@ func newDefaultInvariants() invariants.InvariantRegistry {
 
 	invariantTests.AddInvariantOrDie("apiserver-availability", "kube-apiserver", disruptionlegacyapiservers.NewAvailabilityInvariant())
 
+	invariantTests.AddInvariantOrDie(disruptionpodnetwork.InvariantName, "Network / ovn-kubernetes", disruptionpodnetwork.NewPodNetworkAvalibilityInvariant())
```
Comment (JacobTanenbaum) [inline pkg/defaultinvariants/types.go:74]: Nit: you use the const for the invariant name but not for the JIRA organizations name. Should it be consistent?

File: pkg/invariants/network/disruptionpodnetwork/invariant.go [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+	}
+
+	// our pods tolerate masters, so create one for each of them.
+	nodes, err := pna.kubeClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	numNodes := int32(len(nodes.Items))
+
+	// disable the poller until can get the image sorted out.
```
Comment (JacobTanenbaum) [inline pkg/invariants/network/disruptionpodnetwork/invariant.go:100]: Nit: maybe tag with a `TODO:` so it has less a chance of being lost?

Comment (deads2k) [reply pkg/invariants/network/disruptionpodnetwork/invariant.go:100]: > Nit: maybe tag with a `TODO:` so it has less a chance of being lost?

I personally guarantee I won't lose it :)

File: pkg/invariants/network/disruptionpodnetwork/pod-network-target-deployment.yaml (hunk 1 above)
Comment (JacobTanenbaum) [reply pkg/invariants/network/disruptionpodnetwork/pod-network-target-deployment.yaml:30]: Yeah do a followup

File: pkg/invariants/network/disruptionpodnetwork/invariant.go (hunk 3 above)
Comment (deads2k) [reply pkg/invariants/network/disruptionpodnetwork/invariant.go:93]: added

**Comment by deads2k** [comment]
was lgtmed, simple rebase, tests pass, no conflict, merging.

**Pull request merged**